web/modules/custom/eulogin/src/JoinupEuExternalAuthnameItemList.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_eulogin;
+
+use Drupal\Core\Field\FieldItemList;
+use Drupal\user\Entity\User;
+
+/**
+ * Computed field that returns the external authname of a user.
+ */
+class JoinupEuExternalAuthnameItemList extends FieldItemList {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function createItem($offset = 0, $value = NULL) {
```

Comment:
I have thought of this actually but do we really want this? There is actually a service for this that contains a method that easily does this, the `\Drupal\cas\Service\CasUserManager::setCasUsernameForAccount`. However, when a user is logging in through CAS, more things are happening, like storing of the session data. 
If we were to go with this, we would have to also logout the user in case he is logged in and also clear the data. 
This would be also ran every time a user is saved. Do we really want to interfere with CAS and Authmap by saving a computed field's value?

Reply:
We would also have to maintain tests on that. This is not something that is actionable in the UI as the computed field is not configurable in the form nor does it have any widget, so basically we would maintain unused code. I do not think that we will prefer to use the 
```
$account = User::load(123);
$account->eulogin_authname = 'cas_authname';
$account->save();
```
instead of using the proper service.